lib/src/native/realm_core.dart

```diff
@@ -61,25 +61,25 @@ class _RealmCore {
 
   String get libraryVersion => _realmLib.realm_get_library_version().cast<Utf8>().toDartString();
 
-  LastError? getLastError([Allocator? allocator]) {
-    if (allocator != null) {
-      final error = _realmLib.realm_get_last_error();
-
-      if (error == nullptr) {
-        return null;
-      }
-
-      String? message;
-      if (error.ref.message != nullptr) {
-        message = error.ref.message.cast<Utf8>().toDartString();
-      }
-      _realmLib.realm_release_last_error(error);
+  LastError? getLastError(Arena arena) {
+    final error = arena<realm_error_t>();
+    final success = _realmLib.realm_get_last_error(error);
+    if (!success) {
+      return null;
+    }
 
-      return LastError(error.ref.error, message);
+    String? message;
+    if (error.ref.message != nullptr) {
+      message = error.ref.message.cast<Utf8>().toDartString();
     }
 
-    return using((Arena arena) {
-      return getLastError(arena);
+    return LastError(error.ref.error, message);
+  }
+
+  void raiseLastErrorAsRealmException([String? errorMessage]) {
+    using((Arena arena) {
+      final lastError = getLastError(arena);
+      throw RealmException("${errorMessage ?? ""} ${lastError.toString()}");
```

Comment:
```suggestion
      throw RealmException("${errorMessage ?? ""} Error: ${lastError.toString()}");
```

Reply:
Recall that `lastError.toString` writes `Error code: ${lastError.code}` potentially followed by `Message: ${lastError.message}` if such exists. But I have added a '. ' between, if `message` is set.

Reply:
👍 resolving